### ISSUE TITLE
cURL returns weird JSON

### ISSUE THREAD

**Issue by iriiiina** [validator/validator#465]
I want to get JSON response about my page. In Postman I use following URL and it works fine (returns 2 errors about AngularJS tags):
`http://validator.w3.org/check?uri=https://irina-ivanova.gitlab.io&output=json`

But when I try to use same URL with cURL I get weird JSON response with strange formatting:
` curl -H "Accept: application/json" "https://validator.w3.org/check?uri=https://irina-ivanova.gitlab.io&output=json"`

Response:

    Content-Type: application/json; charset=UTF-8
    X-W3C-Validator-Recursion: 1
    X-W3C-Validator-Status: Invalid
    X-W3C-Validator-Errors: 2
    X-W3C-Validator-Warnings: 1

    {
        "url": "https://irina-ivanova.gitlab.io/",
        "messages": [
        
              {
              
              
                  "message": ,
                  "messageid": "html5",
                  "explanation": "    <p class=\"helpwanted\">\n      <a\n        href=\"feedback.html?uri=https%3A%2F%2Firina-ivanova.gitlab.io%2F;errmsg_id=html5#errormsg\"\n\ttitle=\"Suggest improvements on this error message through our feedback channels\" \n      >&#x2709;</a>\n    </p>\n",
                  "type": "info"
              },
        
          {
              "lastLine": 207,
              "lastColumn": 34,
              "message": "Bad value {{p.url}} for attribute href on element a: Illegal character in path segment: { is not allowed.",
              "messageid": "html5",
              "explanation": "    <p class=\"helpwanted\">\n      <a\n        href=\"feedback.html?uri=https%3A%2F%2Firina-ivanova.gitlab.io%2F;errmsg_id=html5#errormsg\"\n\ttitle=\"Suggest improvements on this error message through our feedback channels\" \n      >&#x2709;</a>\n    </p>\n",
              "type": "error"
          },
        
          {
              
              
              "message": ,
              "messageid": "html5",
              "explanation": "    <p class=\"helpwanted\">\n      <a\n        href=\"feedback.html?uri=https%3A%2F%2Firina-ivanova.gitlab.io%2F;errmsg_id=html5#errormsg\"\n\ttitle=\"Suggest improvements on this error message through our feedback channels\" \n      >&#x2709;</a>\n    </p>\n",
              "type": "info"
          },
        
          {
              "lastLine": 208,
              "lastColumn": 86,
              "message": "Bad value {{p.url}} for attribute href on element a: Illegal character in path segment: { is not allowed.",
              "messageid": "html5",
              "explanation": "    <p class=\"helpwanted\">\n      <a\n        href=\"feedback.html?uri=https%3A%2F%2Firina-ivanova.gitlab.io%2F;errmsg_id=html5#errormsg\"\n\ttitle=\"Suggest improvements on this error message through our feedback channels\" \n      >&#x2709;</a>\n    </p>\n",
              "type": "error"
          },
        
          {
              
              
              "message": ,
              "messageid": "html5",
              "explanation": "    <p class=\"helpwanted\">\n      <a\n        href=\"feedback.html?uri=https%3A%2F%2Firina-ivanova.gitlab.io%2F;errmsg_id=html5#errormsg\"\n\ttitle=\"Suggest improvements on this error message through our feedback channels\" \n      >&#x2709;</a>\n    </p>\n",
              "type": "info"
          },
        
          {
              
              
              "message": ,
              "messageid": "html5",
              "explanation": "    <p class=\"helpwanted\">\n      <a\n        href=\"feedback.html?uri=https%3A%2F%2Firina-ivanova.gitlab.io%2F;errmsg_id=html5#errormsg\"\n\ttitle=\"Suggest improvements on this error message through our feedback channels\" \n      >&#x2709;</a>\n    </p>\n",
              "type": "info"
          },
        
          {
              
              
              "message": ,
              "messageid": "html5",
              "explanation": "    <p class=\"helpwanted\">\n      <a\n        href=\"feedback.html?uri=https%3A%2F%2Firina-ivanova.gitlab.io%2F;errmsg_id=html5#errormsg\"\n\ttitle=\"Suggest improvements on this error message through our feedback channels\" \n      >&#x2709;</a>\n    </p>\n",
              "type": "info"
          }
        
        ],
    "source": {
        "encoding": "utf-8",
        "type": "text/html"
    }
    }

Am I using cURL incorrectly or it's a bug?

**Comment by Arttse** [comment]
Hi @iriiiina!

Try this:
```sh
$ curl -H "Accept: application/json" "https://validator.w3.org/nu/?doc=https%3A%2F%2Firina-ivanova.gitlab.io&out=json"
```

Output:
```
{"url":"https://irina-ivanova.gitlab.io/","messages":[{"type":"error","lastLine":207,"lastColumn":34,"firstColumn":17,"message":"Bad value “{{p.url}}” for attribute “href” on element “a”: Illegal character in path segment: “{” is not allowed.","extract":"d=\"title\"><a href={{p.url}}>{{p.ti","hiliteStart":10,"hiliteLength":18},{"type":"error","lastLine":208,"lastColumn":86,"firstColumn":69,"message":"Bad value “{{p.url}}” for attribute “href” on element “a”: Illegal character in path segment: “{” is not allowed.","extract":" 200}}... <a href={{p.url}}>Read m","hiliteStart":10,"hiliteLength":18}],"language":"en"}
```

**Comment by iriiiina** [comment]
Thanks, that helped! (I took first link from https://validator.w3.org/docs/api.html)

**Comment by sideshowbarker** [comment]
> Thanks, that helped! (I took first link from https://validator.w3.org/docs/api.html)

Thanks for catching that—I’ve updated https://validator.w3.org/docs/api.html to correct it
